runtime/doc/lsp.txt

```diff
@@ -2249,6 +2249,35 @@ stop({bufnr}, {client_id})                    *vim.lsp.semantic_tokens.stop()*
       • {client_id}  (`integer`) The ID of the |vim.lsp.Client|
 
 
+==============================================================================
+Lua module: vim.lsp.on_type_formatting                *lsp-on_type_formatting*
+
+start({bufnr}, {client_id})               *vim.lsp.on_type_formatting.start()*
```

Comment:
Although dealing with a lot of formatting providers can be funky, we do handle multiple clients in `vim.lsp.buf.format` and I don't think we should differ from that here.

Reply:
I think we should keep this separate -- distinguishing "do $THING _now" and "set up autocommands for $THING to happen automatically" are quite different things.

Not opposed to `vim.lsp.buf.format_enable` though. (Should be called `enable` to match completion and LSP in general.)